README.md

```diff
@@ -8,7 +8,121 @@ the latest version of the library via the CI system. To interactively generate
 different visualizations of the data collected here, check out
 [Ginkgo Performance Explorer](https://ginkgo-project.github.io/gpe).
 
-## License
 
-TBD: (Don't do anything commercial with the data until then. Using it for
-research is fine.)
+## Repository structure and metadata
+
+This repository contains two types of information:
+1. Benchmark data of the Ginkgo library on different hardware in the [`data`
+   folder](data);
+2. Plot scripts for the GPE in the [`plots` folder](plots).
+
+For most use cases, only benchmark `data` (1) are relevant.
+
+### Repository structure
+
+In the [`data/` folder](data), you can find the actual benchmark data listed.
+The data is organized in a hierarchy of folders, with the following levels:
+
+1. The hardware that is benchmarked, e.g. `MI100`
+2. The Ginkgo executor controlling that hardware, e.g. `hip`
+3. The type of data used by the benchmark, e.g. `SuiteSparse` for matrices from
+   the [SuiteSparse Collection](https://sparse.tamu.edu/), or `blas.json` for
+   synthetic Dense Linear Algebra benchmarks.
+4. In some cases, extra levels are provided as data classification. For
+   `SuiteSparse`, the matrices are put into different directories based on the
+   collection they belong to.
+5. The final benchmark data is always in standalone benchmark files.
+
+
+Note that aggregated benchmark data can be present in the root `data` folder,
+but they are only convenience files for the Ginkgo Performance Explorer and are
+not always up to date. Scripts are provided also in the main `data` folder to
+aggregate the standalone SuiteSparse JSON files.
+
+Most of the data can be found in the `master` branch. Data can also be found in
+other branches, either because the data was uploaded for debugging purposes, or
+in the context of a scientific paper.
+
+### Commits and data sources
+
+The data can be added by:
+1. The @ginkgo-bot account;
+2. Any user which wants to share his Ginkgo data benchmarks.
```

Comment:
Maybe we should think about a PR format structure, so that external users who want to push may have a PR instead of directly pushing to the main branch ?

Reply:
I think we could yes. We definitely need to come up with a way to clean up this repository, we have many branches and generally have made somewhat of a mess, it's hard to figure out where to find which data.